bson/binary.py

```diff
@@ -247,6 +247,11 @@ def __init__(self, data: Sequence[float | int], dtype: BinaryVectorDtype, paddin
         self.dtype = dtype
         self.padding = padding
 
+    def __repr__(self) -> str:
+        return "BinaryVector(dtype={}, padding={}, data={})".format(  # noqa: UP032
```

Comment:
> will __repr__ create the string even when it's not called?

f-string vs format makes no difference. If anything f-string will be faster. `__repr__` only runs when repr() is called.

Reply:
> `__` only runs when repr() is called.

So if one has logger.debug(vector), is it not true that str.format won't be called, but the f-string will be created?

Reply:
Logging behavior is unrelated to this PR since we're just adding a new method.